ecs-agent/tmds/handlers/fault/v1/handlers/handlers.go

```diff
@@ -203,23 +218,82 @@ func (h *FaultHandler) CheckNetworkBlackHolePort() func(http.ResponseWriter, *ht
 		rwMu.RLock()
 		defer rwMu.RUnlock()
 
-		// TODO: Check status of current fault injection
-		// TODO: Return the correct status state
-		responseBody := types.NewNetworkFaultInjectionSuccessResponse("running")
-		logger.Info("Successfully checked status for fault", logger.Fields{
-			field.RequestType: requestType,
-			field.Request:     request.ToString(),
-			field.Response:    responseBody.ToString(),
-		})
+		ctx := context.Background()
+		ctxWithTimeout, cancel := context.WithTimeout(ctx, requestTimeoutDuration)
+		defer cancel()
+
+		var responseBody types.NetworkFaultInjectionResponse
+		var statusCode int
+		chainName := fmt.Sprintf("%s-%s-%s", *request.TrafficType, *request.Protocol, strconv.FormatUint(uint64(*request.Port), 10))
+		running, cmdOutput, cmdErr := h.checkNetworkBlackHolePort(ctxWithTimeout, request, chainName, taskMetadata.TaskNetworkConfig.NetworkNamespaces[0].Path)
+
+		// We've timed out trying to check if the black hole port fault injection is running
+		if err := ctx.Err(); err != nil {
+			statusCode = http.StatusInternalServerError
+			responseBody = types.NewNetworkFaultInjectionErrorResponse(requestTimedOutError)
+		} else if cmdErr != nil {
+			statusCode = http.StatusInternalServerError
+			responseBody = types.NewNetworkFaultInjectionErrorResponse(cmdOutput)
+		} else {
+			statusCode = http.StatusOK
+			if running {
+				responseBody = types.NewNetworkFaultInjectionSuccessResponse("running")
+			} else {
+				responseBody = types.NewNetworkFaultInjectionSuccessResponse("not-running")
+			}
+			logger.Info("Successfully checked status for fault", logger.Fields{
+				field.RequestType: requestType,
+				field.Request:     request.ToString(),
+				field.Response:    responseBody.ToString(),
+			})
+		}
+
 		utils.WriteJSONResponse(
 			w,
-			http.StatusOK,
+			statusCode,
 			responseBody,
 			requestType,
 		)
 	}
 }
 
+// checkNetworkBlackHolePort will check if there's a running black hole port within the task network namespace based on the chain name and the passed in required request fields.
+// It does so by calling iptables linux utility tool.
+func (h *FaultHandler) checkNetworkBlackHolePort(ctx context.Context, request types.NetworkBlackholePortRequest, chain, netNs string) (bool, string, error) {
+	cmdString := fmt.Sprintf(iptablesChainExistCmd, chain, *request.Protocol, strconv.FormatUint(uint64(*request.Port), 10))
+	cmdList := strings.Split(cmdString, " ")
+
+	// For host mode, the task network namespace is the host network namespace (i.e. we don't need to run nsenter)
+	if netNs != "host" {
+		cmdList = append(strings.Split(fmt.Sprintf(nsenterCommandString, netNs), " "), cmdList...)
+	}
+
+	cmdOutput, err := h.runExecCommand(ctx, cmdList)
+	if err != nil {
+		if strings.HasPrefix(string(cmdOutput), iptablesChainNotFoundError) {
```

Comment:
Dumb question - can we get exit status from the cmd output?

When given chain is found, the exit status code should be 0 otherwise it will be 1.

```
[root@ip-10-194-20-22 ~]# iptables -nL
Chain INPUT (policy ACCEPT)
target     prot opt source               destination         

Chain FORWARD (policy ACCEPT)
target     prot opt source               destination         

Chain OUTPUT (policy ACCEPT)
target     prot opt source               destination         

Chain Xing (0 references)
target     prot opt source               destination         
DROP       tcp  --  0.0.0.0/0            0.0.0.0/0            tcp dpt:123
[root@ip-10-194-20-22 ~]# iptables -C Xing -j DROP -p tcp --dport 123
[root@ip-10-194-20-22 ~]# echo $?
0
[root@ip-10-194-20-22 ~]# iptables -C Xing -j DROP -p tcp --dport 1234
iptables: Bad rule (does a matching rule exist in that chain?).
[root@ip-10-194-20-22 ~]# echo $?
1
[root@ip-10-194-20-22 ~]#
```

Reply:
fyi this is what I'm used for local testing
```
package main

import (
        "fmt"
        "os/exec"
)

func main() {
        cmd := exec.Command("iptables", "-C", "Xing", "-j", "DROP", "-p", "tcp", "--dport", "1234")
        err := cmd.Run()

        if err != nil {
                // Check if the error is an ExitError
                if exitErr, ok := err.(*exec.ExitError); ok {
                        // Get the exit code
                        exitCode := exitErr.ExitCode()
                        fmt.Println("Exit Code:", exitCode)
                } else {
                        // Handle other errors
                        fmt.Println("Error:", err)
                }
        } else {
                fmt.Println("Command executed successfully")
        }
}
```

Reply:
Seems like we only get the exit status from the error that's returned.

In my local testing, this is what I logged
```
stdErr="iptables: Bad rule (does a matching rule exist in that chain?).\n" stdOut="" err="exit status 1
```

Reply:
It also returns an exit status code `1` if `iptables` is not found IIRC

Reply:
Yes, you are correct that the exit status code is from the error

Reply:
Discussed offline, better to not match by certain standard error output/messages